services/question/README.md

```diff
@@ -0,0 +1,103 @@
+# Question Service User Guide
+
+## Get All Questions
+This endpoint allows the retrieval of all the questions in the database.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions`
+
+### Responses:
+
+| Response Code | Explanation                                         |
+|---------------|-----------------------------------------------------|
+| 200 (OK)      | Success, all questions are returned                 |
+| 500 (Internal Server Error) | Unexpected error in the database or server |
+
+**Example of Response Body for Success**:  
+TBC
+
+## Get Question by ID
+This endpoint allows the retrieval of the question by using the question ID.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/{questionId}`
+
+### Parameters:
+
+- Required: `questionId` path parameter
+
+### Responses:
+
+| Response Code | Explanation                                         |
+|---------------|-----------------------------------------------------|
+| 200 (OK)      | Success, question corresponding to the questionID is returned |
+| 404 (Not Found) | Question with the specified questionID not found   |
+| 500 (Internal Server Error) | Unexpected error in the database or server |
+
+**Example of Response Body for Success**:  
+TBC
+
+## Get Question by Parameters
+This endpoint allows the retrieval of a random question that matches the parameters provided.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/search`
+
+### Parameters:
+
+- `limit` - The number of questions to be returned (Required)
```

Comment:
Doesn't this endpoint retrieve a single question?

Reply:
Oh the description I wrote was a bit misleading. I was thinking we could just have set it by default to 1 for finding a question for a user. Then for values other than 1, I was thinking we could have it just in case there's some other usage for it (eg; retrieval of questions that matches the param for admin to look through). But I guess the randomised nature of this wouldn't really work that well for the example I gave LOL unless there's another boolean flag to indicate if we want random or not.

Reply:
yep sounds good, we could have optional parameters `topics`, `difficulty`, `limit` in the get all questions endpoint for table pagination and filtering 😄